### PR TITLE
adds filterMap to challenge

### LINKED ISSUE #7: add filterMap function
Since most of this challenge is just educational on how to use reduce, it would be nice to have a function which is actually useful using the methodologies they've used above.

A filterMap is a good example of this. It's a function which behaves the same as
```js
array.filter(filterFn).map(mapFn)
```
but doesn't iterate over the array twice, by using reduce.

### PR DESCRIPTION
**Pull request by ZooeyMiller**
Adds filterMap challenge, tests and solution

fix #7

### CODE REVIEW

**Review by mantagen — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -69,10 +69,26 @@ function join(array, string) {
...
+//
+// This is a function which takes an array, a function which returns a boolean,
+// and a function which maps the array element.
+//
+// It should be functionally the same as array.filter(filterFn).map(mapFn)
+//
+// This is actually a useful function to write with reduce as it allows you to
+// filter and map an array without having to iterate over it twice.
+// For those interested, the type signature for this function is:
+// [a] -> (a -> Bool) -> (a -> b) -> [b]
```
Comment (mantagen) [inline index.js]: Could be
filterMap :: [a] -> (a -> Bool) -> ...etc

Basically just add the thing with the function name and the double colon.
Then it looks more familiarly like a type signature.

File: index.js [hunk 2]
```diff
@@ -69,10 +69,26 @@ function join(array, string) {
   //your code here
 }
 
+// Now let's make something new entirely! We're going to make filterMap.
+//
+// This is a function which takes an array, a function which returns a boolean,
```
Comment (astroash) [inline index.js]: This sentence took me a few tries to understand. How about laying it out in a list?
eg
// The function filterMap takes 3 arguments:
- an array
- a function that returns a boolean,
- a function that maps the array element.

File: index.js [hunk 3]
```diff
@@ -69,10 +69,26 @@ function join(array, string) {
...
+// and a function which maps the array element.
+//
+// It should be functionally the same as array.filter(filterFn).map(mapFn)
+//
+// This is actually a useful function to write with reduce as it allows you to
```
Comment (astroash) [inline index.js]: Maybe highlight that you could use this in production code
eg

Unlike the previous functions, this is actually a...

File: index.js (hunk 1 above)
Comment (astroash) [reply index.js]: I'm also into this 🌵

File: .index.solution.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ function join(array, string) {
   });
 }
 
-function filterMap(array, filterFn, mapFn) {
+function filterMap(filterFn, mapFn, array) {
```
Comment (astroash) [inline .index.solution.js]: FUNctional 🐻

**Review by astroash — APPROVED** [review]
🌮

**Pull request merged**